logstash-core/lib/logstash/environment.rb

```diff
@@ -34,6 +34,7 @@ module Environment
   end
 
   [
+           Setting::Boolean.new("run_as.root", false),
```

Comment:
I find it a bit weird we define the setting as 'run_as.root'. I doubt we will ever have to add other settings under 'run_as', so I would think we should just create a flat config name, e.g.:

'allow_root'

Or similar.

Reply:
Indeed naming is always difficult :) We shortly discussed with @yaauie and making it `open to extend` instead of limiting to `boolean` makes sense for this case. And finalized with following config:
`on_superuser: BLOCK, WARN(default), ALLOW`

Let me please know if still I can improve or missing. Thanks!